homeassistant/components/device_tracker/icloud.py

```diff
@@ -29,6 +29,7 @@ def setup_scanner(hass, config, see):
     # Get the username and password from the configuration
     username = config.get(CONF_USERNAME)
     password = config.get(CONF_PASSWORD)
+    name = config.get('name')
```

Comment:
Also, add `username` as second argument so that that is returned when no config entry for name has been found.
```python
name = config.get(CONF_NAME, username)
```